### ISSUE TITLE
shrake-rupley error quits instead of raising error

### ISSUE THREAD

**Issue by salotz** [mdtraj/mdtraj#1300]
Instead of the `shrake_rupley` function raising an error for particular conditions (atoms are very close together) it simply prints this `ERROR: THIS CODE IS KNOWN TO FAIL WHEN ATOMS ARE VIRTUALLYON TOP OF ONE ANOTHER. YOU SUPPLIED TWO ATOMS 0.000000APART. QUITTING NOW` and quits the whole process.

This should definitely raise an error which I can except and do what I want with it.

The problem is here: https://github.com/mdtraj/mdtraj/blob/8e85405586e549109f8865046867e1fda7f580d6/mdtraj/geometry/src/sasa.cpp#L92

I would fix it myself but I don't know anything about C interfaces.

**Comment by mpharrigan** [comment]
We can change the function to return an error code and check it in the python wrapping

**Comment by salotz** [comment]
That would work. If I can be of help, let me know.

**Comment by rmcgibbo** [comment]
@cxhernandez: lmk if you still need me. this is pretty straightforward to fix.

**Comment by salotz** [comment]
If there is any progress on this on any branch let me know. I would like to use this ASAP, and in the meanwhile I will try to make a fix myself.

**Comment by salotz** [comment]
https://github.com/salotz/mdtraj/tree/sasa_error_raising/mdtraj/geometry
I converted all the C functions to `int` types in place of void and return a -1 instead of exiting. This is handled by the `shrake_rupley` function and raises a `ValueError` if it gets a -1. I got mdtraj to build (using Anaconda: python 3.6.2 and defaults for all the rest of the dependencies),  and compile but I get this error on import:


    In [1]: import mdtraj as mdj
    ---------------------------------------------------------------------------
    RuntimeError                              Traceback (most recent call last)
    __init__.pxd in numpy.import_array()

    RuntimeError: module compiled against API version 0xb but this version of numpy is 0xa

    During handling of the above exception, another exception occurred:

    ImportError                               Traceback (most recent call last)
    <ipython-input-1-2f72b351ccc5> in <module>()
    ----> 1 import mdtraj as mdj

    ~/Dropbox/devel/mdtraj-salotz/mdtraj/__init__.py in <module>()
         27 """
         28 
    ---> 29 from .formats.registry import FormatRegistry
         30 from .formats.xtc import load_xtc
         31 from .formats.trr import load_trr

    ~/Dropbox/devel/mdtraj-salotz/mdtraj/formats/__init__.py in <module>()
          2 
          3 from .arc import ArcTrajectoryFile
    ----> 4 from .dcd import DCDTrajectoryFile
          5 from .binpos import BINPOSTrajectoryFile
          6 from .xtc import XTCTrajectoryFile

    ~/Dropbox/devel/mdtraj-salotz/mdtraj/formats/dcd/dcd.pyx in init dcd()
         30 import numpy as np
         31 cimport numpy as np
    ---> 32 np.import_array()
         33 from mdtraj.utils import ensure_type, cast_indices, in_units_of
         34 from mdtraj.utils.six import string_types

    __init__.pxd in numpy.import_array()

    ImportError: numpy.core.multiarray failed to import

**Comment by smiles724** [comment]
Hi, I met the same problem and discovered that it is still not completely solved. 
The program directly quitted rather than reporting an error information. 
![image](https://github.com/mdtraj/mdtraj/assets/39299863/b6555bab-cfa1-46dd-9c84-6be22e176d0e)
information.

**Comment by salotz** [comment]
I think the wontfix label should be removed.

**Comment by sukritsingh** [comment]
Done!

**Comment by JavierSanchez-Utges** [comment]
Yes, and this issue should be open, no? It was raised 7 years ago, and does not seem fixed yet...

**Comment by sukritsingh** [comment]
Oh yeah, whoops. Let me go ahead and do that (although I confess I have to dig into the code to see what might be going on here)
